### ISSUE TITLE
cost 15 request produces cost 14 result

### ISSUE THREAD

**Issue by roycewilliams** [uberswe/bcrypt_fun#5]
If cost 15 is requested, a cost 14 hash is generated.

(And hey, long time no talk - hope you're well!)

**Comment by uberswe** [comment]
Hey @roycewilliams, nice to see that you are using this 😃 

The label for the text field states that a number between 0 and 15 is valid. The reason I put this limit is that it just takes too long (in my opinion) for anything higher than a cost of 14 to be generated.

<img width="577" alt="4151808a4be35d2604d4e6e98dc55a17" src="https://user-images.githubusercontent.com/1903284/111289171-2e256480-8645-11eb-8745-22ab5a890339.png">

The code in question is a simple if statement https://github.com/uberswe/bcrypt_fun/blob/master/api.go#L57-L59

I know that with time we will probably see higher and higher costs as it adds better security so it might be good to re-test this to see how long it takes to generate hashes with higher costs.

Ultimately there will need to be a limit and maybe I can make this clearer with red text or some kind of error notification if a cost which is too high is entered. Do you have any suggestions on what the preferred behaviour would be here?

**Comment by roycewilliams** [comment]
Totally understood that 15 may be too slow for your tastes! Though do be aware that cost 15 is seen in the wild on occasion.

Either way, it sounds like this may just be a UX bug. :) If the actual values allowed are 1 through 14, it may be colloquially less ambiguous to say "Set the cost between 1 and 14" (because most English speakers will take this to mean "1 through 14 inclusive"). Or maybe rephrase as "Costs 1 through 14 are supported" or something like that.

Ah, and looking more closely at your reply, some kind of "cost higher than 14 detected; falling back to cost 14" might be useful UI feedback as well.

Thanks for the tool - hope it's as fun for you as it is for us. :D

**Comment by uberswe** [comment]
This is fixed now, I raised the cost to 20 which takes about 1 minute to generate. It seems to max out a CPU core for the full minute for every request which is made so I made #7 to limit this.
